app/services/OktaUserService.scala

```diff
@@ -0,0 +1,52 @@
+package services
+
+import com.okta.sdk.client.AuthorizationMode.PRIVATE_KEY
+import com.okta.sdk.client.Clients
+import com.okta.sdk.resource.api.UserApi
+import model.User
+import play.api.libs.ws.WSClient
+import play.api.{Configuration, Logging}
+import services.FutureHelper.*
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.CollectionConverters.*
+import scala.util.Failure
+import scala.util.chaining.*
+
+class OktaUserService(config: Configuration, ws: WSClient)(implicit ctx: ExecutionContext)
+    extends UserService
+    with Logging {
+
+  private lazy val orgUrl = s"https://${config.get[String]("oktaApi.domain")}"
+
+  private lazy val userApi = {
+    val client = Clients
+      .builder()
+      .setOrgUrl(orgUrl)
+      .setClientId(config.get[String]("oktaApi.clientId"))
+      .setAuthorizationMode(PRIVATE_KEY)
+      .setPrivateKey(config.get[String]("oktaApi.privateKey"))
+      .setScopes(config.get[Seq[String]]("oktaApi.scopes").toSet.asJava)
+      .build()
+    new UserApi(client)
+  }
+
+  def healthCheck(): Future[Unit] =
+    ws
+      .url(s"$orgUrl/.well-known/openid-configuration")
+      .get()
+      .map(_ => ())
+      .tap(_.onComplete {
+        case Failure(exception) => logger.error(s"Health check failed: ${exception.getMessage}")
+        case _                  => ()
+      })
+
+  def fetchUserByIdentityId(id: String): Future[Option[User]] =
+    tryAsync(
+      userApi
+        .listUsers(null, null, 1, null, s"""profile.legacyIdentityId eq "$id"""", null, null)
```

Comment:
are there any performance / rate limit concerns about searching users in this way ?
The current implementation finds the legacy user first and then just gets the okta user by okta id which might be a more lightweight call I suspect ? (I didn't look it up though)

Reply:
I just looked at it. It seems the rate limit is a lot lower for this type of call than for a regular user lookup by id (and it will compete with data lake exports as well)

Reply:
I haven't done any detailed perf tests but that reminds me I was going to do a comparison.  I just wonder how this would work in the future but it's looking increasingly likely we're going to need some kind of central ID lookup system at least: Okta ID, internal Identity ID, Braze ID, etc.

Both the ID lookup and the user-list calls seem to fall into the same rate limit bucket in the Okta API.

Reply:
it's confusing because the limits are different depending on the query string params. If you use the search query string param the limits are much lower.
To make sure you can use the [okta rate limit headers ](https://developer.okta.com/docs/reference/rl-best-practices/)in the response to check the rate limits to the actual request you made. I think you can check the `X-Rate-Limit-Limit` header in the response you get from okta

Reply:
Now searching for users by Okta ID in f3097e79f46afcfcdd980f317cd6c7bae04b6862.  We do the Postgres search first and then use the Okta ID from that result to do the Okta lookup.